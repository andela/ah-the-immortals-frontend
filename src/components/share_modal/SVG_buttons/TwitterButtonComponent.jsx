import React, {Component} from 'react';
import PropTypes from 'prop-types';

class TwitterButtonComponent extends Component {
  static propTypes = {
    article: PropTypes.object.isRequired
  };
  render() {
    const {article: {slug}} = this.props;
    const url = process.env.URL;
    const currentUrl  = `${url}/post/${slug}`;
    const shareUrl = `https://twitter.com/intent/tweet?text=${currentUrl}`;
    return (
      <div className="svg">
        <button className="btn btn-link" id="twt" type="button" onClick={() => window.open(shareUrl, '_blank')}>
          <img src="https://img.icons8.com/color/64/000000/twitter-circled.png" alt="avatar" />
        </button>
        <div className="twitter-style">Twitter</div>
      </div>
    );
  }
}

export default TwitterButtonComponent;
