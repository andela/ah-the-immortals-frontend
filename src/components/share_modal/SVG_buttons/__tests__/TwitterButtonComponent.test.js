import { mount } from 'enzyme';
import React from 'react';
import TwitterButtonComponent from '../TwitterButtonComponent';



describe('Test for the twitter button component', () =>{
  const props = {
    article: {
      slug: ''
    }
  };
  const wrapper = mount(
    <TwitterButtonComponent {...props} />
  );
  it('Test the mounting of the facebook button', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Simulates clicking of open button',()=>{
    const openButton = wrapper.find('button').first();
    openButton.simulate('click',
      global.open = jest.fn()
    );
    expect(global.open).toBeCalled();
  });
});
