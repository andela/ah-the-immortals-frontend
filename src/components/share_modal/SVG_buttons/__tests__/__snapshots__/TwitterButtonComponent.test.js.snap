// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the twitter button component Test the mounting of the facebook button 1`] = `ReactWrapper {}`;
